canonical/vanilla-framework
# Update the snapshots script to use combined examples if they're found

## Done

First segment of split-up of #5117 as requested by @pastelcyborg .

This PR only includes the changes to `snapshots.js` to make it snapshot `combined.html` templates where found, and take snapshots in every color theme on every example. Note that until combined examples themselves are merged, this script would mean a dramatic increase in snapshots per build as we are taking snapshots in each theme now.

It also cleans up `snapshots.js` slightly to make sure it is well-documented, and adds a set of Jest tests to verify correctness of the Snapshots script.

## QA

- Verify PR test passes (it will run a Jest test against `snapshots.test.js` verifying correctness of the Percy script)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
